ground/gcs/src/plugins/coreplugin/mainwindow.cpp

```diff
@@ -138,6 +130,31 @@ MainWindow::MainWindow() :
 #endif /* Q_OS_MAC */
     m_toggleFullScreenAction(0)
 {
+    // keep this in sync with main() in app/main.cpp
+    m_settings = new QSettings(XmlConfig::XmlSettingsFormat, QSettings::UserScope,
```

Comment:
> not that it is bad to have the autosaved settings hanging around.

This. When I wrote this code, I had no expectation that it would work so robustly. So I kept around an autosave because recreating all my settings was a giant PITA as well as impediment to workflow. In retrospect, since implementing this change I haven't had a single settings corruption and so making the autosave probably isn't necessary.